components/core/src/clp/streaming_archive/writer/utils.cpp

```diff
@@ -25,17 +25,19 @@ auto split_file(
         Archive& archive_writer
 ) -> void {
     auto const& encoded_file = archive_writer.get_file();
-    auto has_ts_pattern = encoded_file.has_ts_pattern();
-    auto orig_file_id = encoded_file.get_orig_file_id();
-    auto split_ix = encoded_file.get_split_ix();
+    auto const has_ts_pattern = encoded_file.has_ts_pattern();
+    auto const orig_file_id = encoded_file.get_orig_file_id();
+    auto const end_message_ix = encoded_file.get_end_message_ix();
+    auto const split_ix = encoded_file.get_split_ix();
     archive_writer.set_file_is_split(true);
     close_file_and_append_to_segment(archive_writer);
 
-    archive_writer.create_and_open_file(path_for_compression, group_id, orig_file_id, ++split_ix);
+    archive_writer.create_and_open_file(path_for_compression, group_id, orig_file_id, split_ix + 1);
     if (has_ts_pattern) {
         // Initialize the file's timestamp pattern to the previous split's pattern
         archive_writer.change_ts_pattern(last_timestamp_pattern);
     }
+    archive_writer.set_file_begin_message_ix(end_message_ix);
```

Comment:
I think the begin message index should be passed in whenever a file is created and we should set it after creation. Let's make it as a parameter in `create_and_open_file`